### PR TITLE
[Doc] Make doc code snippet testable [5/n]

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Change code snippet from ..code-block:: to ..testcode::

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by angelinalg — APPROVED** [review]
LGTM

File: doc/source/ray-observability/monitoring-debugging/gotchas.rst [hunk 1]
```diff
@@ -52,7 +52,8 @@ runs on another machine it won't.
 
 And I have this code:
 
-.. code-block:: python
+.. testcode::
+  :skipif: True
```
Comment (bveeramani) [inline doc/source/ray-observability/monitoring-debugging/gotchas.rst:56]: Would it make sense to test this code and match any output?

```python
.. testoutput::
    :hide:
    :options: +ELLIPSIS

    ...
```

File: doc/source/ray-observability/monitoring-debugging/gotchas.rst [hunk 2]
```diff
@@ -112,8 +114,12 @@ of Ray Tasks itself, e.g.
...
 
 This will error with message:
-ValueError: Cannot schedule create_task_that_uses_resources.<locals>.sample_task with the placement group
-because the resource request {'CPU': 10} cannot fit into any bundles for the placement group, [{'CPU': 1.0}].
+
+.. testoutput::
+  :options: +SKIP
+
+    ValueError: Cannot schedule create_task_that_uses_resources.<locals>.sample_task with the placement group
+    because the resource request {'CPU': 10} cannot fit into any bundles for the placement group, [{'CPU': 1.0}].
```
Comment (bveeramani) [inline doc/source/ray-observability/monitoring-debugging/gotchas.rst:122]: Turns out it's possible to test exceptions. Would it be possible to test this `ValueError`?

```
.. testcode::

    assert False

.. testoutput::
    :options: +ELLIPSIS

    Traceback (most recent call last):
      ...
    AssertionError
```

File: doc/source/ray-observability/ray-logging.rst [hunk 3]
```diff
@@ -331,7 +341,8 @@ If using the Ray cluster launcher, you would specify this environment variable i
 
 When connecting to the cluster, be sure to set the environment variable before connecting:
 
-.. code-block:: python
+.. testcode::
+    :skipif: True
 
     os.environ["RAY_LOG_TO_STDERR"] = "1"
     ray.init(address="auto")
```
Comment (bveeramani) [inline doc/source/ray-observability/ray-logging.rst:348]: Would it be possible to test this?

**Review by bveeramani — APPROVED** [review]
LGTM!

**Review by bveeramani — APPROVED** [review]
LGTM!

File: doc/source/ray-observability/monitoring-debugging/gotchas.rst (hunk 2 above)
Comment (jjyao) [reply doc/source/ray-observability/monitoring-debugging/gotchas.rst:122]: Not for this case. The error message is in `error.txt` generated by `tune` not in stdout.

File: doc/source/ray-observability/ray-logging.rst (hunk 3 above)
Comment (jjyao) [reply doc/source/ray-observability/ray-logging.rst:348]: Probably hard: we need to start a ray cluster via `ray start` first.

**Pull request merged**